src/controllers/auth.controller.js

```diff
@@ -0,0 +1,185 @@
+  });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+  await user.save();
+
+  res.send(user);
+};
+// як перенаправити на профіль????
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await User.findOne({ where: { email } });
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  generateToken(res, user);
+  ConsoleLoger.log(`${user.email} login`);
+};
+// як перенаправити на профіль????
+
+const refresh = async (req, res) => {
+  const { refreshToken } = req.cookies;
+
+  const userData = await jwtService.verifyRefresh(refreshToken);
+  const token = await jwtService.signRefresh(refreshToken);
+
+  if (!userData || !token) {
+    throw ApiError.unauthorize();
+  }
+
+  const email = userData.email;
+
+  const user = await User.findOne({ where: { email } });
+
+  generateToken(res, user);
+};
+
+const generateToken = async (res, user) => {
+  const normalizeUser = userService.normalize(user);
+
+  const accessToken = await jwtService.sign(normalizeUser);
+  const refreshToken = await jwtService.signRefresh(normalizeUser);
+
+  await tokenService.save(normalizeUser.id, refreshToken);
+
+  res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    httpOnly: true,
+  });
+
+  res.send({
+    user: normalizeUser,
+    accessToken,
+  });
+};
+
+const logout = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.verifyRefresh(refreshToken);
+
+  if (!userData || !refreshToken) {
+    throw ApiError.unauthorize();
+  }
+
+  await tokenService.remove(userData.id);
+
+  res.clearCookie('refreshToken');
+
+  res.sendStatus(204);
+};
+
+const reset = async (req, res) => {
+  const { email } = req.body;
+  const user = User.findOne({ where: { email } });
+
+  if (!user) {
+    throw ApiError.badRequest('Not found');
+  }
+
+  userService.sendResetEmail(email);
+
+  ConsoleLoger.log(email);
+  res.send('Email send, check your email for reset password');
+};
+
+const resetPassword = async (req, res) => {
+  const { resetToken } = req.params;
+  const { newPassword, confirmation } = req.body;
+
+  const user = await User.findOne({ where: { resetToken } });
+
+  if (!user || !resetToken) {
+    throw ApiError.badRequest('Invalid reset token or user not found');
+  }
+
+  if (newPassword !== confirmation) {
+    res.status(400).send('Passwords do not match');
+
+    return;
+  }
+
+  const hashPass = await bcrypt.hash(newPassword, 10);
+
+  userService.changePassword(user.userName, hashPass);
+  res.send('Password change sucssessfull');
```

Comment:
There is a typo in 'Password change sucssessfull'. It should be 'Password change successful'.